### PR TITLE
Ensure we take compiler-provided declaration of posix_memalign

### PR DESCRIPTION
**Pull request by asl**
Some systems (e.g. those with GNU libc) declare posix_memalign with an exception specifier. Compilers (clang including) might have special handling of this to allow posix_memalign redeclaration with / without exception specifier. As we override posix_memalign for GC purposes we really need to ensure the proper include order to workaround this weirdness.

### CODE REVIEW

**Comment by fruffy** [comment]
Thanks, we can merge this instead of https://github.com/p4lang/p4c/pull/4595. They are doing similar things.

**Comment by asl** [comment]
@fruffy Yes, I am checking different approaches and how the problem does manifest.

**Review by fruffy — APPROVED** [review]
Conditionally approving assuming `COMPILE_WITH_CLANG: ON` is removed.

**Comment by asl** [comment]
@fruffy How about adding builder that would build p4c using clang on Ubuntu?

**Comment by vlstill** [comment]
> @fruffy How about adding builder that would build p4c using clang on Ubuntu?

Considering the discussion I had with @fruffy regarding the static builds, this should be probably part of nightly builds not PR suite (the risk of breakage would be relatively small I guess, especially breakage just under clang and not on macOS). I think having a clang build in CI would be useful.

**Comment by fruffy** [comment]
We do already have a Clang nightly build, it is the sanitizer one. It just does not use unity builds.

**Comment by asl** [comment]
> We do already have a Clang nightly build, it is the sanitizer one. It just does not use unity builds.

Ah, great. How does it do notifications?

**Comment by fruffy** [comment]
> > We do already have a Clang nightly build, it is the sanitizer one. It just does not use unity builds.
> 
> Ah, great. How does it do notifications?

Notifications are kind of strange here: https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/notifications-for-workflow-runs

The last user that modified the workflow (me) gets notified of failures.

File: lib/gc.cpp
Comment (asl) [inline lib/gc.cpp:21]:
```suggestion
// ensure the proper include order to workaround this weirdness.
```

**Comment by asl** [comment]
@fruffy Ok, I think this is the least bad approach, still not perfect, but we'd have to live with this as-is. Merging when ready.

**Comment by fruffy** [comment]
> @fruffy Looks like ptf-ebpf fails due to some network issue?

Looks like a spurious failure, I restarted the test. It's optional so it should not block merging anyway.

**Comment by asl** [comment]
> Looks like a spurious failure, I restarted the test. It's optional so it should not block merging anyway.

I already restarted it 3 or 4 times...

**Comment by fruffy** [comment]
> > Looks like a spurious failure, I restarted the test. It's optional so it should not block merging anyway.
> 
> I already restarted it 3 or 4 times...

Ah just saw. Hmm, there is nothing we can really do about it for now. I would ignore it. If the issue persists we can start to look into it. This particular test is not super important.

**Pull request merged**
